overlord/snapstate/target.go

```diff
@@ -607,15 +682,39 @@ func (p *pathInstallGoal) toInstall(ctx context.Context, st *state.State, opts O
 		return nil, err
 	}
 
+	comps, err := installableComponentsFromPaths(info, p.components)
+	if err != nil {
+		return nil, err
+	}
+
 	inst := target{
-		setup: &SnapSetup{
+		setup: SnapSetup{
 			SnapPath:  p.path,
 			Channel:   channel,
 			CohortKey: p.revOpts.CohortKey,
 		},
-		info:   info,
-		snapst: snapst,
+		info:       info,
+		snapst:     snapst,
+		components: comps,
 	}
 
 	return []target{inst}, nil
 }
+
+func installableComponentsFromPaths(info *snap.Info, components map[*snap.ComponentSideInfo]string) ([]ComponentSetup, error) {
+	setups := make([]ComponentSetup, 0, len(components))
+	for csi, path := range components {
+		compInfo, _, err := backend.OpenComponentFile(path, info, csi)
```

Comment:
shouldn't this do something more similar to validatedInfoFromPathAndSideInfo ?

Reply:
Yes, resolved in 301c185f92b91b21466af283bbeba04436a40fb4.